### PR TITLE
Completes 7. POST/api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by Tayamul**

### CODE REVIEW

File: controllers/articles.controller.js [hunk 1]
```diff
@@ -34,4 +35,19 @@ const getArticlesCommentsById = (req, res, next) => {
     .catch((err) => next(err));
 };
 
-module.exports = { getArticles, getArticlesById, getArticlesCommentsById };
+const postComments = (req, res, next) => {
+  const { article_id } = req.params;
+
+  insertComments(req.body, article_id)
+    .then((comments) => {
+      res.status(201).send({ comments });
```
Comment (B7Bprog) [inline controllers/articles.controller.js]: You should only send back the newly posted comment here and it should be an object.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -187,31 +187,171 @@ describe("GET/api/articles/:article_id/comments", () => {
...
+
+describe("POST/api/articles/:article_id/comments", () => {
+  test("201: post a new comment on the given article", () => {
+    const article_id = 2;
+    const newComment = {
+      username: "butter_bridge",
+      body: "Being fit matters",
+    };
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
```
Comment (B7Bprog) [inline __tests__/app.test.js:227]: The API should return only the newly posted comment, which is an object, not an array. Please change this test to check for that object and make sure the new comment has the correct ID.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -187,31 +187,171 @@ describe("GET/api/articles/:article_id/comments", () => {
...
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("400: username requested by the client is not in the database", () => {
```
Comment (B7Bprog) [inline __tests__/app.test.js]: This needs to be a 404 'not found'.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -187,31 +187,171 @@ describe("GET/api/articles/:article_id/comments", () => {
...
+      username: "peter_griffin",
+      body: "Being fit matters",
+    };
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
```
Comment (B7Bprog) [inline __tests__/app.test.js]: Please write another test (404) where the article ID is valid data type, but there's no article with that ID.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Tayamul, please see my comments, amend your code and add tests. Then please send us another pull request with the update.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -187,31 +187,171 @@ describe("GET/api/articles/:article_id/comments", () => {
...
+      .expect(201)
+      .then(({ body: { comments } }) => {
+        expect(comments).toBeInstanceOf(Array);
+        expect(comments).toHaveLength(1);
+        comments.forEach((comment) => {
+          expect(comment).toEqual(
+            expect.objectContaining({
+              comment_id: expect.any(Number),
+              body: expect.any(String),
+              article_id: expect.any(Number),
```
Comment (popatre) [inline __tests__/app.test.js]: article id is 2

File: __tests__/app.test.js [hunk 6]
```diff
@@ -187,31 +187,171 @@ describe("GET/api/articles/:article_id/comments", () => {
...
+              author: expect.any(String),
```
Comment (popatre) [inline __tests__/app.test.js]: author: "butter_bridge",
      body: "Being fit matters",

File: __tests__/app.test.js [hunk 7]
```diff
@@ -187,31 +187,171 @@ describe("GET/api/articles/:article_id/comments", () => {
...
+    };
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("400: missing a username from the client's request (value)", () => {
```
Comment (popatre) [inline __tests__/app.test.js]: 404 -username not found

File: __tests__/app.test.js [hunk 8]
```diff
@@ -187,31 +187,171 @@ describe("GET/api/articles/:article_id/comments", () => {
...
+    };
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("400: PSQL would not allow to post comments on a non-existent article, as it violates foreign key constraint", () => {
```
Comment (popatre) [inline __tests__/app.test.js]: 404 id not found

File: __tests__/app.test.js (hunk 3 above)
Comment (popatre) [inline __tests__/app.test.js]: 404 -username not found

File: __tests__/app.test.js (hunk 7 above)
Comment (popatre) [reply __tests__/app.test.js]: Same as bottom one

File: controllers/articles.controller.js (hunk 1 above)
Comment (popatre) [inline controllers/articles.controller.js]: key of comment

**Review by popatre — CHANGES_REQUESTED** [review]

**Review by B7Bprog — APPROVED** [review]
Nice one Tayamul, please merge it! 🎉

**Pull request merged**